EndlessClient/UIControls/CharacterInfoPanel.cs

```diff
@@ -168,6 +174,12 @@ protected virtual async Task DeleteButtonClick()
         protected virtual void DoUpdateLogic(GameTime gameTime)
         {
             _characterControl.Update(gameTime);
+
+            var keyboardState = _userInputProvider.CurrentKeyState;
+            if (keyboardState.IsKeyDown((Microsoft.Xna.Framework.Input.Keys)49+_characterIndex))
```

Comment:
I think you want to check currentState.KeyUp and previousState.KeyDown for this, otherwise holding the key will spawn a bunch of tasks. The interlocked guard happens in `LoginButtonClick` so it won't stop the tasks from being started (and then immediately exiting).

Reply:
Will do :)

Reply:
I switched to using IsKeyPressedOnce